### PR TITLE
Add unnified logging system tests for macOS

### LINKED ISSUE #1330: Adapt Remoted simulator to gather macOS ULF events
### Issue information
| Epic 	| QA base branch 	| Wazuh branch 	|
|-	|-	|-	|
| https://github.com/wazuh/wazuh-qa/issues/1329 	| master 	|  4572-dev-uls-logs-enhancement |

## Description

Is necessary to create new tests for unified logging system (ULS) macOS events research about the following:

### Investigate Remoted simulator

Remoted simulator is a wazuh-qa module designed to create an AF_INET server socket for simulating remoted connection. It is required to research if this tool could be useful to gather ULS events from the Wazuh macOS agent and if so, adapting it for use in logcollector tests.

  
### Adapt macOS mock tool

[macOS mock tool](https://github.com/wazuh/wazuh-tools/tree/master/utils/macos_log) is designed to replicate macOS log stream tool in GNU/Linux environment. It could be desirable to adapt it in order to use it in logcollector tests.

### PR DESCRIPTION
**Pull request by Rebits**
|Related issue|
|---|
|Closes #1330 |

## Description

This PR adds new tools to test macOS unified logging system(ULS) events. Also, a basic test has been created to ensure Wazuh macOS agent sends correctly ULS events.

## Implementation

In order to gather Wazuh macOS ULS events sent by the agent, remoted simulator has been adapted to store sent events in a queue. 

Untitled Diagram

To generate ULS events in the agent host, we have created some functions `macos_os_log_message` and `macos_logger_message` that uses `logger` (https://www.unix.com/man-page/osx/1/logger/) and `os_log` (https://developer.apple.com/documentation/os/os_log) function respectively.

- **logger  ULS event**

logger_message

- **os_log event**

custom_log_console

### MacOS

test_macos_results

## Documentation
documentation-macos

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/logcollector.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 import sys
-from os import path
 from math import ceil
 from json import load
 from time import sleep
 from datetime import datetime, timedelta
-
+from tempfile import gettempdir
+from wazuh_testing.tools import WAZUH_PATH
 from wazuh_testing.tools import LOGCOLLECTOR_STATISTICS_FILE, monitoring
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/logcollector.py:10]:
```suggestion
from wazuh_testing.tools import LOGCOLLECTOR_STATISTICS_FILE, WAZUH_PATH, monitoring
```

File: deps/wazuh_testing/wazuh_testing/logcollector.py [hunk 2]
```diff
@@ -19,6 +19,19 @@
...
+    'client_keys': os.path.join(WAZUH_PATH, 'etc', 'client.keys'),
+    'server_keys': os.path.join(WAZUH_PATH, 'etc', 'sslmanager.keys'),
+    'server_cert': os.path.join(WAZUH_PATH, 'etc', 'sslmanager.cert'),
+    'authd_port': 1515,
+    'remoted_port': 1514,
+    'protocol': 'tcp',
+    'remoted_mode': 'CONTROLLED_ACK',
+}
+
+TEMPLATE_OSLOG_MESSAGE = 'Custom os_log event message'
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/logcollector.py:33]:
```suggestion
TEMPLATE_MACOS_LOG_MESSAGE = 'Custom os_log event message'
```

File: deps/wazuh_testing/wazuh_testing/logcollector.py [hunk 3]
```diff
@@ -546,10 +559,62 @@ def wait_statistics_file():
...
+    Args:
+        type (str): Log type (info, debug, default, error or fault).
+        subsystem (str): Subsystem of the event log.
+        category (str): Category of the event log.
+        process_name (str): Name of the process that is going to generate the log.
+    """
+    compiled_log_generator_path = os.path.join(gettempdir(), process_name)
+    if not os.path.exists(compiled_log_generator_path):
+        os_log_swift_script = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                                           'tools/macos_log/log_generator.swift')
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/logcollector.py:597]:
```suggestion
                                           'tools', 'macos_log', 'log_generator.swift')
```

File: docs/tests/integration/test_logcollector/index.md [hunk 4]
```diff
@@ -46,4 +47,9 @@ For each location and exclude option specified in the configuration file, check
 Wazuh allows forwarding the events that are written in a monitored log file to a UNIX `named socket` 
 through the `target` option in the `localfile` section of the configuration. These tests will check 
 if the statistics of the forwarded events, which are in the file `wazuh-logcollector.state` are correct, 
-verifying that the dropped events match with the reported ones.
\ No newline at end of file
+verifying that the dropped events match with the reported ones.
+
+#### Test macOS
+
+Wazuh macOS agent allows gather unnified logging system events. These tests ensure logcollector works correctly with this
```
Comment (BraulioV) [inline docs/tests/integration/test_logcollector/index.md:54]:
```suggestion
Wazuh macOS agent allows gathering unified logging system events. These tests ensure logcollector works correctly with this
```

File: docs/tests/integration/test_logcollector/test_macos/index.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Test macOS
+
+## Overview 
+
+Wazuh macOS agent allow gather unnified logging system events. These tests ensure logcollector works correctly with this
```
Comment (BraulioV) [inline docs/tests/integration/test_logcollector/test_macos/index.md:5]:
```suggestion
Wazuh macOS agent allows gathering unified logging system events. These tests ensure logcollector works correctly with this
```

File: docs/tests/integration/test_logcollector/test_macos/test_macos_format_basic.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# Test macOS format basic
+## Overview 
+
+Check if `wazuh-logcollector` gather correctly unnified logging system events.
```
Comment (BraulioV) [inline docs/tests/integration/test_logcollector/test_macos/test_macos_format_basic.md:4]:
```suggestion
Check if `wazuh-logcollector` correctly gathers unified logging system events.
```

This typo is duplicated through the docs. You will need to change all of them.

File: deps/wazuh_testing/wazuh_testing/logcollector.py [hunk 7]
```diff
@@ -546,10 +559,62 @@ def wait_statistics_file():
...
+    """
+    os.system(f"logger {message}")
+
+
+def macos_os_log_message(type, subsystem, category, process_name="custom_log"):
+    """Create a unified logging system log using log generator script.
+
+    To create a custom event log with desired type, subsystem and category the `log_generator` script is required.
+    This, get these parameters and use os_log (https://developer.apple.com/documentation/os/os_log) to create it.
+    To run correctly `log_generator` is necessary to compile it. This is done in temporal folder, using `process_name`
```
Comment (spothound) [inline deps/wazuh_testing/wazuh_testing/logcollector.py:585]:
```suggestion
    To correctly run  `log_generator` is necessary to compile it. This is done in a temporal folder, using `process_name`
```

File: deps/wazuh_testing/wazuh_testing/tools/manager_keys.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
+sslmanager_key = """-----BEGIN PRIVATE KEY-----
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/tools/manager_keys.py:1]: This file can be moved to the `data` directory and load them in the `tools` module.

File: deps/wazuh_testing/wazuh_testing/logcollector.py [hunk 9]
```diff
@@ -546,10 +559,62 @@ def wait_statistics_file():
...
+def macos_logger_message(message):
+    """Create a unified logging system log using logger tool.
+
+    Args:
+        message (str): Logger event message.
+    """
+    os.system(f"logger {message}")
+
+
+def macos_os_log_message(type, subsystem, category, process_name="custom_log"):
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/logcollector.py:580]: Be careful with the name of the functions. They don't represent what they're doing and also the name can be confusing.

File: deps/wazuh_testing/wazuh_testing/tools/remoted_sim.py [hunk 10]
```diff
@@ -107,38 +112,47 @@ def _start_socket(self):
...
         self.encryption_key = sum2 + sum1
 
     def compose_sec_message(self, message, binary_data=None):
-        """
-        Compose event from raw message
+        """Compose event from raw message.
+
+        Args:
+            message (str): Raw message.
+            binary_data(str): Binary data.
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/tools/remoted_sim.py:155]:
```suggestion
            binary_data (str): Binary data.
```
Also, this doc string... Is a bit obvious. Can you give more details about this variable?

**Review by BraulioV — CHANGES_REQUESTED** [review]

File: tests/integration/test_logcollector/conftest.py [hunk 11]
```diff
@@ -21,3 +23,43 @@ def restart_logcollector(get_configuration, request):
...
     setattr(request.module, 'wazuh_log_monitor', file_monitor)
     control_service('start', daemon=DAEMON_NAME)
+
+
+@pytest.fixture(scope='module')
+def init_authd_remote_simulator(get_connection_configuration, request):
+    """Initialize authd and remoted simulator
+
+    Args:
+        get_connection_configuration (fixture): Dictionary with authd and remoted parameters.
```
Comment (BraulioV) [inline tests/integration/test_logcollector/conftest.py:33]: `request` is missing

File: deps/wazuh_testing/wazuh_testing/tools/__init__.py [hunk 12]
```diff
@@ -84,7 +84,7 @@ def get_service():
     return 'wazuh-manager' if service == 'server' else 'wazuh-agent'
 
 
-_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+_data_path = os.path.join(os.path.join(os.path.dirname(os.path.realpath(__file__)),'..'), 'data')
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/tools/__init__.py:87]:
```suggestion
_data_path = os.path.join(os.path.join(os.path.dirname(os.path.realpath(__file__)), '..'), 'data')
```

**Review by BraulioV — CHANGES_REQUESTED** [review]

**Review by BraulioV — APPROVED** [review]

**Pull request merged**